cmd/eval_test.go

```diff
@@ -1312,6 +1312,108 @@ time.clock(input.y, time.clock(input.x))
 	}
 }
 
+func TestEvalDiscardOutput(t *testing.T) {
+	tests := map[string]struct {
+		query, format, expected string
+		params                  evalCommandParams
+	}{
+		"success example": {
+			query: "1*2+3",
+			params: func() evalCommandParams {
+				params := newEvalCommandParams()
+				err := params.outputFormat.Set(evalDiscardOutput)
+				if err != nil {
+					t.Fatalf("unexpected error: %s", err)
+				}
+				return params
+			}(),
+			expected: `{
+  "result": "discarded"
+}
+`},
+		"error example": {
+			query: "1/0",
+			params: func() evalCommandParams {
+				params := newEvalCommandParams()
+				err := params.outputFormat.Set(evalDiscardOutput)
+				if err != nil {
+					t.Fatalf("unexpected error: %s", err)
+				}
+				return params
+			}(),
+			expected: `{}
+`},
+		"error example show built-in-errors": {
+			query: "1/0",
+			params: func() evalCommandParams {
+				params := newEvalCommandParams()
+				err := params.outputFormat.Set(evalDiscardOutput)
+				if err != nil {
+					t.Fatalf("unexpected error: %s", err)
+				}
+				params.showBuiltinErrors = true
+				return params
+			}(),
+			expected: `{
+  "errors": [
+    {
+      "code": "eval_builtin_error",
+      "location": {
+        "col": 1,
+        "file": "",
+        "row": 1
+      },
+      "message": "div: divide by zero"
+    }
+  ]
+}
+`},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			var buf bytes.Buffer
+			_, err := eval([]string{tc.query}, tc.params, &buf)
+			if err != nil {
+				t.Fatalf("unexpected error: %s", err)
+			}
+			if actual := buf.String(); actual != tc.expected {
+				t.Errorf("expected output %q\ngot %q", tc.expected, actual)
+			}
+		})
+	}
+}
+
+func TestEvalDiscardProfilerOutput(t *testing.T) {
```

Comment:
suggestion: here are some minor improvements I'd make to this test, namely I'd unpack the output in a different way, but make fewer assertions about the profile. I think we just need to check that the field is still there (rather than making any other assertions on the shape of the data in there)

```go
func TestEvalDiscardProfilerOutput(t *testing.T) {
	params := newEvalCommandParams()
	err := params.outputFormat.Set(evalDiscardOutput)
	if err != nil {
		t.Fatalf("unexpected error: %s", err)
	}
	params.profile = true

	query := "1*2+3"

	var buf bytes.Buffer
	_, err = eval([]string{query}, params, &buf)
	if err != nil {
		t.Fatalf("unexpected error: %s", err)
	}

	var output map[string]interface{}
	if err := util.NewJSONDecoder(&buf).Decode(&output); err != nil {
		t.Fatal(err)
	}

	// assert tha the result is set to discarded
	result, ok := output["result"].(string)
	if !ok {
		t.Fatal("error extracting result as string from output")
	}

	if result != "discarded" {
		t.Fatal("Expected result field to be set to 'discarded'")
	}

	// assert that profile is still set
	_, ok = output["profile"]
	if !ok {
		t.Fatal("error in parsing profile output")
	}
}
```

This is just a suggestion - up to you 🙂

Reply:
That makes sense, thank you 🙂

Reply:
Glad to hear it 🙂 think we're nearly there.